src/app/qgsmaptoolscalefeature.cpp

```diff
@@ -0,0 +1,479 @@
+      if ( pointGeometry.isNull() )
+      {
+        return;
+      }
+
+      double minDistance = std::numeric_limits<double>::max();
+
+      QgsFeature cf;
+      QgsFeature f;
+      while ( fit.nextFeature( f ) )
+      {
+        if ( f.hasGeometry() )
+        {
+          double currentDistance = pointGeometry.distance( f.geometry() );
+
+          if ( currentDistance < minDistance )
+          {
+            minDistance = currentDistance;
+            cf = f;
+          }
+        }
+      }
+
+      if ( minDistance == std::numeric_limits<double>::max() )
+      {
+        emit messageEmitted( tr( "Could not find a nearby feature in the current layer." ) );
+        return;
+      }
+
+      mExtent = cf.geometry().boundingBox();
+      mFeatureCenter = mExtent.center();
+
+      mScaledFeatures.clear();
+      mScaledFeatures << cf.id(); //todo: take the closest feature, not the first one...
+
+      mRubberBand = createRubberBand( mLayer->geometryType() );
+      mRubberBand->setToGeometry( cf.geometry(), mLayer );
+    }
+    else
+    {
+      mScaledFeatures = mLayer->selectedFeatureIds();
+
+      mRubberBand = createRubberBand( mLayer->geometryType() );
+
+      QgsFeature feat;
+      QgsFeatureIterator it = mLayer->getSelectedFeatures();
+      while ( it.nextFeature( feat ) )
+      {
+        mRubberBand->addGeometry( feat.geometry(), mLayer );
+      }
+    }
+    QgsPointXY mapAnchor = toMapCoordinates( mLayer,mFeatureCenter );
+    QPoint rubberAnchor = toCanvasCoordinates( mapAnchor );
+    mAnchorPoint->setCenter( mapAnchor );
+    mRubberScale = QPointF( rubberAnchor.x() - mRubberBand->x(), rubberAnchor.y() - mRubberBand->y() );
+    mRubberBand->setTransformOriginPoint( rubberAnchor );
+    mRubberBand->show();
+    mBaseDistance = toLayerCoordinates( mLayer, e->mapPoint() ).distance( mFeatureCenter );
+    mScaling = 1.0;
+
+    createScalingWidget();
+    if ( e->modifiers() & Qt::ShiftModifier )
+    {
+      if ( mScalingWidget )
+      {
+        mScalingWidget->setMagnet( 45 );
+      }
+    }
+
+    mScalingActive = true;
+
+    return;
+  }
+
+  applyScaling( mScaling );
+}
+
+void QgsMapToolScaleFeature::cancel()
+{
+  deleteScalingWidget();
+  deleteRubberband();
+  QgsVectorLayer *mLayer = currentVectorLayer();
+  if ( mLayer->selectedFeatureCount() == 0 )
+  {
+    mAnchorPoint.reset();
+  }
+  mScalingActive = false;
+}
+
+void QgsMapToolScaleFeature::updateRubberband( double scale )
+{
+  if ( mScalingActive )
+  {
+    mScaling = scale;
+
+    double offsetx = ( 1 - mScaling ) * mRubberScale.x();
+    double offsety = ( 1 - mScaling ) *  mRubberScale.y();
+
+    if ( mRubberBand )
+    {
+      mRubberBand->setTransform( QTransform( mScaling, 0, 0, mScaling, offsetx, offsety ) );
+      mRubberBand->update();
+    }
+  }
+}
+
+
+void QgsMapToolScaleFeature::applyScaling( double scale )
+{
+  mScaling = scale;
+  mScalingActive = false;
+
+  QgsVectorLayer *mLayer = currentVectorLayer();
```

Comment:
```suggestion
  QgsVectorLayer *vlayer = currentVectorLayer();
```

Only member variables should use the m prefix

Reply:
removed that assignation, using cached layer.